### PR TITLE
Integrate Additional Logging of Information Received Over DroneCAN

### PR DESCRIPTION
**Pull request by vertiq-jordan**
This PR brings in support for the logging of additional DroneCAN information. First, it allows for the logging of DroneCAN nodes' Node Status information. At the moment, PX4 accepts and monitors node statuses, but it is helpful to be able to review additional information after a flight. Next, I've introduced support for the (fairly new) DroneCAN ESC Status Extended message. Like DroneCAN ESC Status, the Status Extended message provides information about connected DroneCAN ESCs. The extended message adds input and output throttle percentages, motor temperature in degrees Celsius, the ESC index reporting, and additional vendor specific status flags. This PR adds support for both receiving and logging information received by ESCs that report the Status Extended message.

Here are a couple of logs that include logged Node Status and Status Extended from 3 connected ESCs. 
The first is an application of logging these two data sets as part of the rest of the basic logging topics: [extended_and_node_status_demo.zip](https://github.com/user-attachments/files/17603281/extended_and_node_status_demo.zip)

The second has configured the SD card to only log the DroneCAN ESC Status Extended and DroneCAN Node Status messages
[extended_and_node_status_logged_only.zip](https://github.com/user-attachments/files/17603285/extended_and_node_status_logged_only.zip)

### CODE REVIEW

File: src/drivers/uavcan/libdronecan/libuavcan/include/uavcan/protocol/node_status_monitor.hpp [hunk 1]
```diff
@@ -10,6 +10,8 @@
 #include <uavcan/node/subscriber.hpp>
 #include <uavcan/node/timer.hpp>
 #include <uavcan/protocol/NodeStatus.hpp>
+#include <uORB/Publication.hpp>
+#include <uORB/topics/dronecan_node_status.h>
```
Comment (dakejahl) [inline src/drivers/uavcan/libdronecan/libuavcan/include/uavcan/protocol/node_status_monitor.hpp:14]: We probably don't want to couple uORB into the dronecan library. Can you make an interface instead and add the uORB plumbing to **uavcan_main.cpp/hpp**

Comment (vertiq-jordan) [reply src/drivers/uavcan/libdronecan/libuavcan/include/uavcan/protocol/node_status_monitor.hpp]: Alrighty, I've taken all uorb references out of the dronecan library, and all publishing has been offloaded to uavcan main

**Comment by dakejahl** [comment]
Hey Jordan thanks for this. I've taken a closer look at this and I think I see a much cleaner way to do this:

1. There already exists an `Entry` struct in `NodeStatusMonitor`, let's just complete the implementation by adding `uptime_sec` and `vendor_specific_status_code` to this struct.

2. In uavcan_main where you added your logic you can instead call `_node_status_monitor.forEachNode` to then iterate over, advertise, and publish the status info for each online node. For example:
https://github.com/PX4/PX4-Autopilot/blob/main/src/drivers/uavcan/uavcan_main.cpp#L1132-L1136

3. This will require using `uORB::PublicationMulti` so that each node status gets its own instance of `DronecanNodeStatusData` and you can delete the `DronecanNodeStatus` message.

With your current implementation there's the issue that if uavcan_main does not loop quickly enough it will only get the node status from the most recently received message.

**Comment by dakejahl** [comment]
I broke out the NodeStatus portion of this into a separate PR https://github.com/PX4/PX4-Autopilot/pull/23890

**Comment by vertiq-jordan** [comment]
> I broke out the NodeStatus portion of this into a separate PR #23890. Can you cherry pick the ESC stuff from here into a new PR?

Will do!

**Comment by vertiq-jordan** [comment]
ESC Status Extended portion of this PR is out now @dakejahl, so I

**Comment by vertiq-jordan** [comment]
ESC Status Extended portion of this PR is out now @dakejahl, so I'm going to kill this one off since we have both parts handled now!

**Pull request closed without merging**